File: src/plugin/combat/magic/ancient/teleport/PaddewwaTeleportSpell.java
package plugin.combat.magic.ancient.teleport;

import com.rs.game.entity.WorldTile;
import com.rs.game.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 7/27/2017
 */
public class PaddewwaTeleportSpell extends TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 54;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(LAW_RUNE, 2, FIRE_RUNE, 1, AIR_RUNE, 1);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(3099, 9882, 0);
	}
	
	@Override
	public int spellId() {
		return 40;
	}
	
	@Override
	public double exp() {
		return 64;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.ANCIENTS;
	}
}

File: src/main/java/plugin/interaction/combat/magic/ancient/teleport/PaddewwaTeleportSpell.kt
package plugin.interaction.combat.magic.ancient.teleport

import game.content.plugin.combat.spell.type.TeleportSpellPlugin
import game.global.WorldTile
import utility.constants.MagicConstants
import utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 7/27/2017
 */
class PaddewwaTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 54
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.LAW_RUNE, 2, MagicConstants.FIRE_RUNE, 1, MagicConstants.AIR_RUNE, 1)
    }

    override fun destination(): WorldTile {
        return WorldTile(3099, 9882, 0)
    }

    override fun spellId(): Int {
        return 40
    }

    override fun exp(): Double {
        return 64.0
    }

    override fun book(): MagicBook {
        return MagicBook.ANCIENTS
    }
}